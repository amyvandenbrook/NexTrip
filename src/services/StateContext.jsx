import React, { createContext, useReducer } from 'react'
import Reducer from './Reducer.jsx'

const StateContext = createContext()

const StateProvider = ({ children, initialState }) => {
    const defaultState = {
        routeData: [],
        route: undefined,
        routeChanged: false,
        directionData: [],
        direction: '',
        stopData: [],
    }
    const [state, dispatch] = useReducer(Reducer, {
        ...defaultState,
        ...initialState,
    })

    return (
        <StateContext.Provider
            value={{
                ...state,
                dispatch,
            }}
        >
            {children}
        </StateContext.Provider>
    )
}

export { StateContext, StateProvider }
