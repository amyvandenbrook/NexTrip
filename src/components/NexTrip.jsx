import React from 'react'
import { StateProvider } from '../services/StateContext.jsx'
import Direction from './Direction'
import Route from './Route'
import Stops from './Stops'
import '../styles/main.scss'

const NexTrip = () => {
    return (
        <StateProvider>
            <div className="content">
                <div className="disclaimer">
                    <p>This application is sourced by real-time data from </p>
                    <img
                        name="Metro Transit"
                        alt="Metro Transit Logo"
                        src="https://www.metrotransit.org/img/MetroTransitLogo.svg"
                    ></img>
                </div>
                <h1 className="header--1">NexTrip</h1>
                <div className="route_container">
                    <Route />

                    <div
                        className="directions_container"
                        data-testid="Direction-Container"
                    >
                        <Direction />
                    </div>
                </div>
                <Stops />
            </div>
        </StateProvider>
    )
}

export default NexTrip
