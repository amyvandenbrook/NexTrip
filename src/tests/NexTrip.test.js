import React from 'react'
import { act, create } from 'react-test-renderer'
import axios from 'axios'
import { StateProvider } from '../services/StateContext.jsx'
import NexTrip from '../components/NexTrip.jsx'

jest.mock('axios')
axios.get = jest.fn()

describe('Stops', () => {
    let nextrip
    let component

    test('generates the application with default data', () => {
        act(() => {
            component = create(
                <StateProvider
                    initialState={{
                        routeData: ['route1'],
                        route: 'mockRoute',
                        direction: 'mockNorth',
                        directionData: [
                            { Text: 'direction1', Value: 'value1' },
                        ],
                        stopData: [{ Text: 'mockStop' }],
                    }}
                >
                    <NexTrip />
                </StateProvider>
            )
        })
        nextrip = component.toJSON()
        expect(nextrip).toMatchSnapshot()
    })
})
