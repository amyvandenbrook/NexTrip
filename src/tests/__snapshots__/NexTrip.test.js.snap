// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stops generates the application with default data 1`] = `
<div
  className="content"
>
  <div
    className="disclaimer"
  >
    <p>
      This application is sourced by real-time data from 
    </p>
    <img
      alt="Metro Transit Logo"
      name="Metro Transit"
      src="https://www.metrotransit.org/img/MetroTransitLogo.svg"
    />
  </div>
  <h1
    className="header--1"
  >
    NexTrip
  </h1>
  <div
    className="route_container"
  >
    <p>
      Please select a route from the options below to see the directions and stops available for you!
    </p>
    <select
      className="route_selector"
      data-testid="Select-Route"
      id="Route"
      name="Route"
      onChange={[Function]}
    >
      <option
        value={-1}
      >
        Select a Route
      </option>
      ;
       
    </select>
    <i
      className="fas fa-arrow-circle-down"
    />
    <div
      className="directions_container"
      data-testid="Direction-Container"
    />
  </div>
</div>
`;
